### PR TITLE
Add group finder cron job

### LINKED ISSUE #253: Remove groups from Algolia that were deleted in Rock

### LINKED ISSUE #254: Group preference resolves null

### PR DESCRIPTION
**Pull request by jordanwade**
### 1. What does this PR do, or why is it needed?

Adds a cron job using Apollos core setup to sync/index groups to Algolia on a regular interval.

**Minor fixes/refinements (from Ryan)**
- Deletes all group records from Algolia _right_ before adding the new ones (Closes #253)
- Fixes the `preferences` method to properly fall back on default value when `titleOverride` isn't set (Fixes #254)
- Updates logging and developer friction so it's not too easy to accidentally trigger indexes locally
- Cleans up the linting in Group Item data source and the Algolia search-index files.
  - Most of the "real" changes are in [these commits](https://github.com/christfellowshipchurch/apollos-api/pull/256/files/178bec509503c9949a7b478e1a208aa9fd15906f) and [these ones
](https://github.com/christfellowshipchurch/apollos-api/pull/256/files/9ee84701f3320a6615d67284aa6f796d605b9c4e..c3ddeb9320da0a6bdec118a6621cb2d0b2f3fef7)

### 2. Goals

- Straight forward simple jobs implementation for Group Finder
- Avoid overage charges on Algolia
- Low friction for getting up to date Groups synced in a timely manner

### 3. How do I test this PR?

Check logs on Bull Dashboard make sure the syncing is working as expected. http://localhost:4000/admin/queues

Monitor Algolia for a week or so to make sure we aren't blowing our operations budget.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes [tag-issues-here]
- [x] No new warnings in tests, storybook, or apps
- [x] Assign relevant reviewers
- [x] Upload GIF(s) of iOS, Android, Web
- [ ] Get one peer approval before merging

## REVIEW:

**Code Review**

- [x] Read through the "Files changed" tab _very carefully_
- [x] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [x] How might we make this easier for someone else to understand?
- [x] Could the code be simpler?
- [x] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: src/data/group-item/jobs.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+const createJobs = ({ getContext, queues, trigger = () => null }) => {
+  const FullIndexQueue = queues.add('algolia-groups-full-index-queue', queueOpts);
+
+  FullIndexQueue.process(async () => {
+    const context = getContext();
+    return context.dataSources.GroupItem.updateIndexAllGroups();
+  });
+
+  FullIndexQueue.add(null, { repeat: { cron: '15 3 * * 1' } });
```
Comment (jordanwade) [inline src/data/group-item/jobs.js:37]: Do we want to update this to run everyday at 3:15am? Currently it will just run once a week at 3:15am

**Review by jordanwade — COMMENTED** [review]
@rkdavidson do we want to leave the safety switch on for this PR and turn it off in another PR?

File: src/data/group-item/jobs.js (hunk 1 above)
Comment (rkdavidson) [reply src/data/group-item/jobs.js]: Good call, fixed in bd35502 (I think?). Gut check me, please 🙃 .

Comment (jordanwade) [reply src/data/group-item/jobs.js]: Checked gut 👍
